data/collector.go

```diff
@@ -44,8 +46,7 @@ type collector struct {
 	cancel            context.CancelFunc
 	captureFunc       CaptureFunc
 	closed            bool
-
-	target *datacapture.Buffer
+	target            datacapture.BufferedWriter
```

Comment:
Just clarifying this change from Buffer -> BufferedWriter is so we can use signalingBuffer in the tests